src/harness/sas_test_harness.py

```diff
@@ -325,50 +358,59 @@ def generateCbsdRecords(registration_requests, grant_requests_list):
     # Check if groupingParam optional parameter present in CBSDRecord then
     # add to registration object.
     if registration_request.has_key('groupingParam'):
-      internal_data['registration'].update({'groupingParam': registration_request['groupingParam']})
+      internal_data['registration'].update({'groupingParam': 
+                         registration_request['groupingParam']})
 
     internal_data['grants'] = []
 
-    # Read all grant records if mulitiple grant records are associated with single cbsd.
+    # Read all grant records if mulitiple grant records are associated with 
+    # single cbsd.
     for grant_request in grant_requests:
       grant_data = {}
-      if grant_request.has_key('id'):
-        grant_data['id'] = grant_request['id']
-      if grant_request.has_key('operationParam'):
-        grant_data['operationParam'] = grant_request['operationParam']
-      if grant_request.has_key('channelType'):
-        grant_data['channelType'] = grant_request['channelType']
+
+      # Auto-generating GrantData 'id' field.
+      grant_data['id'] = 'SAMPLE_ID_{:05}'.format(random.randrange(1, 10 ** 5)) 
+      
+      assert grant_request.has_key('operationParam'), "operationParam does not exist in GrantRequest"
+      grant_data['operationParam'] = grant_request['operationParam']
+
```

Comment:
As of this morning in WG3, 'requestedOperationParam' is also a required field in SAS-SAS exchange in Release 1. Please either add a TODO or copy the operationParam into requestedOperationParam as well.

Reply:
Updated   accordingly